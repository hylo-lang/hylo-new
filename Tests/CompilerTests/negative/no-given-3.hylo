//! no-std

fun check<T>(_ x: T) {}

fun f<where let u: Void>() {
  g()
}

fun g<where let u: Void>() {
  check<Void>(u)
}

fun h() {
  f()
}

public fun main() {
  given Void = ()
  f()
}
