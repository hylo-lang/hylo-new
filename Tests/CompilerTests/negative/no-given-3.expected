negative/no-given-3.hylo:14.3-4: error: no coercion from 'Void ==> [Void]() let -> Void' to '[Void]() let -> Void'
  f()
  ^
negative/no-given-3.hylo:14.3-4: note: no given instance of 'Void' in this scope
  f()
  ^
